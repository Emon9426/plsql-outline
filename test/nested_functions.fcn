/* 多层嵌套函数测试 */
CREATE OR REPLACE FUNCTION level1_function(p_input NUMBER) RETURN NUMBER IS
  v_level1 NUMBER := p_input;
  
  FUNCTION level2_function(p_param NUMBER) RETURN NUMBER IS
    v_level2 NUMBER := p_param * 2;
    
    FUNCTION level3_function(p_param NUMBER) RETURN NUMBER IS
      v_level3 NUMBER := p_param + 10;
      
      FUNCTION level4_function(p_param NUMBER) RETURN NUMBER IS
        v_level4 NUMBER;
      BEGIN
        v_level4 := p_param * 3;
        RETURN v_level4;
      END level4_function;
      
    BEGIN
      RETURN level4_function(v_level3);
    END level3_function;
    
  BEGIN
    RETURN level3_function(v_level2);
  END level2_function;
  
BEGIN
  RETURN level2_function(v_level1);
END level1_function;
/
