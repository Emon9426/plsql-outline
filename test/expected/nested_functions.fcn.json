{
  "nodes": [
    {
      "label": "level1_function",
      "type": "function",
      "icon": "symbol-function",
      "children": [
        {
          "label": "level2_function",
          "type": "function",
          "icon": "symbol-function",
          "children": [
            {
              "label": "level3_function",
              "type": "function",
              "icon": "symbol-function",
              "children": [
                {
                  "label": "level4_function",
                  "type": "function",
                  "icon": "symbol-function",
                  "children": [
                    {
                      "label": "BEGIN",
                      "type": "begin",
                      "icon": "symbol-namespace",
                      "children": [],
                      "range": {
                        "startLine": 12,
                        "endLine": 12
                      }
                    }
                  ],
                  "range": {
                    "startLine": 10,
                    "endLine": 15
                  }
                },
                {
                  "label": "BEGIN",
                  "type": "begin",
                  "icon": "symbol-namespace",
                  "children": [],
                  "range": {
                    "startLine": 17,
                    "endLine": 17
                  }
                }
              ],
              "range": {
                "startLine": 7,
                "endLine": 19
              }
            },
            {
              "label": "BEGIN",
              "type": "begin",
              "icon": "symbol-namespace",
              "children": [],
              "range": {
                "startLine": 21,
                "endLine": 21
              }
            }
          ],
          "range": {
            "startLine": 4,
            "endLine": 23
          }
        },
        {
          "label": "BEGIN",
          "type": "begin",
          "icon": "symbol-namespace",
          "children": [],
          "range": {
            "startLine": 25,
            "endLine": 25
          }
        }
      ],
      "range": {
        "startLine": 1,
        "endLine": 27
      }
    }
  ],
  "errors": []
}